### ISSUE TITLE
etcd optional installation for mojaloop and cc, and bringing coredns-nodecache to an independent ansible role

### ISSUE THREAD

**Issue by dfry** [mojaloop/iac-modules#434]
The refactored design involves leveraging this coredns install with some additional pieces:

1. installation of etcd operator
2. install of virtual services for coredns
3. addition of security groups for coredns to allow internal/external traffic to #2
4. iac to generate upstream NS records to point to #2
5. external dns configuration to point to the internal coredns/etcd install

[GitHub - aenix-io/etcd-operator: New generation community-driven etcd-operator!](https://github.com/aenix-io/etcd-operator) 

proposed operator install for etcd, just for supporting coredns as an external dns backend.

also, we need to provide optional changes to expose correct ports in istio as well as load balancer and security groups in order to allow for dns traffic from external networks.

in summary, we need a flag for turning off exposing core dns as public service which would also enable the etcd operator as well as the underlying network access rules.

there would need to be a proof of concept work to show how to setup a zone in route53 that would automatically point to the exposed coredns service and then additionally, setup external dns to point to the local coredns/etcd installation.  that poc work is out of scope for this ticket.
